dask_cuda/utils.py

```diff
@@ -32,14 +32,26 @@ def setup(self, worker=None):
 
 
 class RMMSetup:
-    def __init__(self, nbytes, managed_memory, log_directory):
+    def __init__(self, nbytes, managed_memory, async_alloc, log_directory):
         self.nbytes = nbytes
         self.managed_memory = managed_memory
+        self.async_alloc = async_alloc
         self.logging = log_directory is not None
         self.log_directory = log_directory
 
     def setup(self, worker=None):
-        if self.nbytes is not None or self.managed_memory is True:
+        if self.async_alloc is True:
+            import rmm
+
+            rmm.mr.set_current_device_resource(rmm.mr.CudaAsyncMemoryResource())
+
+            rmm.reinitialize(
+                logging=self.logging,
+                log_file_name=get_rmm_log_file_name(
+                    worker, self.logging, self.log_directory
+                ),
```

Comment:
Should I still be calling `rmm.reinitialize()` in the async case? If so, should `pool_allocator` be enabled?

Reply:
Quoting @shwina : 

> You don’t need the reinitialize call here. reinitialize is a higher-level API to `set_current_device_resource`, but it doesn’t expose the async memory resource. `set_current_device_resource()` should already have “reinitialized” RMM with the pool async MR.

Reply:
Okay, in that case I'll remove that - I guess my larger concern surrounding this is how we would allow for RMM logging when using the async allocator.

Reply:
You could use `rmm.enable_logging()` after calling `set_current_device_resource()`.

Reply:
Great! Thanks for the help 🙂